actors/builtin/power/power_actor.go

```diff
@@ -381,23 +395,34 @@ func (a Actor) processBatchProofVerifies(rt Runtime, rewret reward.ThisEpochRewa
 				infos = append(infos, svi)
 				return nil
 			})
-			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to iterate over proof verify array for miner %s", a)
+			if err != nil {
+				return xerrors.Errorf("failed to iterate over proof verify array for miner %s: %w", a, err)
+			}
 
 			verifies[a] = infos
 			return nil
 		})
-		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to iterate proof batch")
-
+		// Do not return immediately, all runs that get this far should wipe the ProofValidationBatchQueue.
+		// If we leave the validation batch then in the case of a repeating state error the queue
+		// will quickly fill up and repeated traversls will start ballooning cron execution time.
+		if err != nil {
+			stErr = xerrors.Errorf("failed to iterate proof batch: %w", err)
```

Comment:
I think you can simplify this by directly capturing the output of the `ForAll` in `stErr` if you're so inclined

Reply:
I'm going to keep it this way to maintain a standard "wrap err and assign to stErr" pattern.